src/processmoduleconfig/create_config_s390x.go

```diff
@@ -0,0 +1,73 @@
+//go:build s390x
+//+build s390x
+
+package processmoduleconfig
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/spf13/afero"
+)
+
+// createProcessModuleConfigFile creates an empty file at location `destPath` and calls
+// populateProcessModuleConfigFile() to populate the config file for the s390x architecture.
+// This is to fix a bug with the oneagent zip download for s390x where ruxitagentproc.conf is missing from the archive.
+func createProcessModuleConfigFile(fs afero.Fs, destPath string) error {
+	log.Info("generating new ruxitagentproc.conf file")
+	newProcessModuleConfigFile, err := fs.OpenFile(destPath, os.O_WRONLY|os.O_CREATE|os.O_TRUNC, 0644)
+	if err != nil {
+		return err
+	}
+
+	err = populateProcessModuleConfigFile(newProcessModuleConfigFile, destPath)
+	if err != nil {
+		return err
+	}
+
+	return newProcessModuleConfigFile.Close()
+}
+
+// populateProcessModuleConfigFile populates the empty config file on s390x
+func populateProcessModuleConfigFile(cfg afero.File, destPath string) error {
```

Comment:
this logic should be removed as it can't be kept in sync with the actual archive.

Passing the correct architectures to the api should be enough to support ppcle and s390 when they are available on the api.